app/lib/pre_assembly/copy_stager.rb

```diff
@@ -6,6 +6,20 @@ class CopyStager
     def self.stage(source, destination)
       FileUtils.mkdir_p File.dirname(destination)
       FileUtils.cp_r source, destination
+      recursively_chmod_based_on_type!(destination)
+    end
+
+    def self.recursively_chmod_based_on_type!(path)
+      unless File.directory?(path)
+        FileUtils.chmod 0o0664, path
+        return
+      end
+
+      FileUtils.chmod 0o0775, path
+
+      Dir["#{path}/*"].each do |f|
+        recursively_chmod_based_on_type!(f)
+      end
     end
```

Comment:
@andrewjbtw @sul-dlss/infrastructure-team We could replace this entire thing with `FileUtils.chmod_R 'g+rwx,o+rx', destination`, but this has the downside of setting files' executable permission bit, which I think is misleading but not exactly bug-prone or dangerous? 🤷🏻‍♂️

Reply:
depends on if any images or video or audio files have viral executables embedded?  or if any other types come through preassembly???

Reply:
Especially seeing as pre-assembly is a staff only application, I don't know how much we have to worry about about this?

Reply:
I would prefer not to set any files to be executable. It's a staff application but not all content is created by staff - born-digital items may be acquired from outside, for example. 

Also, my hope with this ticket is to get permissions to be consistent with what we already get when creating files using the Preassembly server and the staging space under /dor/staging. Making the files executable would be a whole new thing we haven't done before.

Reply:
@ndushay @peetucket @andrewjbtw OK, thanks, all. I'll leave the permission-wrangling code above as is, then. 😄